### PR TITLE
Pep8 fixes for known_hosts.py

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/system/known_hosts.py

##### ANSIBLE VERSION
```
2.4 devel
```

### CODE REVIEW

**Review by resmo — APPROVED** [review]
shipit

File: lib/ansible/modules/system/known_hosts.py [hunk 1]
```diff
@@ -102,69 +102,68 @@ def enforce_state(module, params):
...
-                                     (path,str(e)))
+                module.fail_json(msg="Failed to read %s: %s" % (path, str(e)))
         try:
             outf = tempfile.NamedTemporaryFile(mode='w+', dir=os.path.dirname(path))
             if inf is not None:
                 for line_number, line in enumerate(inf):
-                    if found_line==(line_number + 1) and (replace_or_add or state=='absent'):
-                        continue # skip this line to replace its key
+                    if found_line == (line_number + 1) and (replace_or_add or state == 'absent'):
+                        continue  # skip this line to replace its key
```
Comment (mcv21) [inline lib/ansible/modules/system/known_hosts.py:157]: I don't like this change - why add a second space before the `#`?

File: lib/ansible/modules/system/known_hosts.py [hunk 2]
```diff
@@ -175,34 +174,32 @@ def enforce_state(module, params):
 
     return params
 
-def sanity_check(module,host,key,sshkeygen):
+
```
Comment (mcv21) [inline lib/ansible/modules/system/known_hosts.py:177]: Why this extra blank line?

File: lib/ansible/modules/system/known_hosts.py [hunk 3]
```diff
@@ -175,34 +174,32 @@ def enforce_state(module, params):
...
         outf.flush()
     except IOError:
         e = get_exception()
-        module.fail_json(msg="Failed to write to temporary file %s: %s" % \
-                             (outf.name,str(e)))
-    rc,stdout,stderr=module.run_command([sshkeygen,'-F',host,
-                                         '-f',outf.name],
-                                        check_rc=True)
+        module.fail_json(msg="Failed to write to temporary file %s: %s" % (outf.name, str(e)))
+    rc, stdout, stderr = module.run_command([sshkeygen, '-F', host, '-f', outf.name], check_rc=True)
```
Comment (mcv21) [inline lib/ansible/modules/system/known_hosts.py]: This is a very long line (and wraps in my editor). Are you sure this is an improvement?

File: lib/ansible/modules/system/known_hosts.py [hunk 4]
```diff
@@ -211,6 +208,7 @@ def sanity_check(module,host,key,sshkeygen):
     if stdout == '':  # host not found
         module.fail_json(msg="Host parameter does not match hashed host field in supplied key")
 
+
```
Comment (mcv21) [inline lib/ansible/modules/system/known_hosts.py:214]: ditto previous comment re extra blank lines before functions.

File: lib/ansible/modules/system/known_hosts.py [hunk 5]
```diff
@@ -323,20 +322,21 @@ def compute_diff(path, found_line, replace_or_add, state, key):
...
-            key       = dict(required=False,  type='str'),
-            path      = dict(default="~/.ssh/known_hosts", type='path'),
-            hash_host = dict(required=False, type='bool' ,default=False),
-            state     = dict(default='present', choices=['absent','present']),
-            ),
-        supports_check_mode = True
-        )
-
-    results = enforce_state(module,module.params)
+        argument_spec=dict(
```
Comment (mcv21) [inline lib/ansible/modules/system/known_hosts.py:332]: No, the previous formatting was much better here for this block, much more readable.

**Comment by mcv21** [comment]
See my comments on the diff. Thanks!

**Comment by mcv21** [comment]
needs_revision

File: lib/ansible/modules/system/known_hosts.py (hunk 1 above)
Comment (Akasurde) [reply lib/ansible/modules/system/known_hosts.py:157]: Inline comment requires 2 spaces.

File: lib/ansible/modules/system/known_hosts.py (hunk 2 above)
Comment (Akasurde) [reply lib/ansible/modules/system/known_hosts.py:177]: pep8 warning say : E302 expected 2 blank lines, found 1

File: lib/ansible/modules/system/known_hosts.py (hunk 3 above)
Comment (Akasurde) [reply lib/ansible/modules/system/known_hosts.py]: Will revert to original.

File: lib/ansible/modules/system/known_hosts.py (hunk 5 above)
Comment (Akasurde) [reply lib/ansible/modules/system/known_hosts.py:332]: ok.

**Comment by Akasurde** [comment]
ready_for_review

**Comment by mcv21** [comment]
Thanks for this.
shipit

File: lib/ansible/modules/system/known_hosts.py [hunk 6]
```diff
@@ -323,20 +325,21 @@ def compute_diff(path, found_line, replace_or_add, state, key):
...
+            name      = dict(required=True, type='str', aliases=['host']),
+            key       = dict(required=False, type='str'),
             path      = dict(default="~/.ssh/known_hosts", type='path'),
-            hash_host = dict(required=False, type='bool' ,default=False),
-            state     = dict(default='present', choices=['absent','present']),
-            ),
-        supports_check_mode = True
-        )
+            hash_host = dict(required=False, type='bool', default=False),
+            state     = dict(default='present', choices=['absent', 'present']),
```
Comment (sivel) [inline lib/ansible/modules/system/known_hosts.py]: If you are fixing pep8 please remove all of these spaces so we can remove this file from legacy.

I see there was a comment here before saying someone liked the formatting better originally but you seem to have squashed commits removing history.

Unfortunately, if that is what the other comment was for, it was wrong.

**Review by sivel — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/system/known_hosts.py (hunk 6 above)
Comment (mcv21) [reply lib/ansible/modules/system/known_hosts.py]: It's a matter of aesthetics, but I do think the `argument_spec` is clearly laid out and very readable as it is (i.e. with all the `=` lining up). Which is why I requested it be left alone.

**Review by sivel — APPROVED** [review]

**Comment by sivel** [comment]
Merged, Thanks!

**Comment by Akasurde** [comment]
@sivel @resmo @mcv21 Thanks.

**Pull request merged**